app/domain/authentication/authn_jwt/fetch_provider_uri_signing_key.rb

```diff
@@ -0,0 +1,53 @@
+module Authentication
+  module AuthnJwt
+    # This class is responsible for fetching JWK Set from provider-uri
+    class FetchProviderUriSigningKey < FetchSigningKeyInterface
+
+      def initialize(authenticator_parameters,
+                     logger,
+                     fetch_required_secrets,
+                     resource_class)
+        @logger = logger
+        @resource_id = authenticator_parameters.authenticator_resource_id
+        @fetch_required_secrets = fetch_required_secrets
+        @resource_class = resource_class
+        @fetch_provider_key = Authentication::OAuth::FetchProviderKeys.new
```

Comment:
why this one is different than the others ? we do not want to pass it as input to the ctor?